docs/guides/commands/application-commands.md

```diff
@@ -0,0 +1,216 @@
+| Version | The version of this interaction |
+| CreatedAt | The time this interaction was created |
+| IsValidToken | Whether or not the token to respond to this interaction is still valid |
+| RespondAsync | Responds to the interaction | 
+| FollowupAsync | Sends a followup message to the interaction |
+
+
+
+#### Whats the difference between `FollowupAsync` and `RespondAsync`?
+RespondAsync is the initial responce to the interaction, its used to "capture" the interaction, while followup is used to send more messages to the interaction.
+Basically, you want to first use `RespondAsync` to acknowledge the interaction, then if you need to send anything else regarding that interaction you would use `FollowupAsync`
+If you have `AlwaysAcknowledgeInteractions` set to true in your client config then it will automatically acknowledge the interaction without sending a message, 
+in this case you can use either or to respond.
+
+#### Example ping pong command
+```cs
+_client.InteractionCreated += MyEventHandler;
+_client.Ready += CreateCommands
+
+...
+
+private async Task CreateCommands()
+{
+  await _client.Rest.CreateGlobalCommand(new Discord.SlashCommandCreationProperties()
+  {
+    Name = "ping",
+    Description = "ping for a pong!",
+  });
+}
+
+private async Task MyEventHandler(SocketInteraction arg)
+{
+  switch(arg.Type) // We want to check the type of this interaction
+  {
+    case InteractionType.ApplicationCommand: // If it is a command
+      await MySlashCommandHandler(arg); // Handle the command somewhere
+      break;
+     default: // We dont support it
+      Console.WriteLine("Unsupported interaction type: " + arg.Type);
+      break;
+  }
+}
+
+private async Task MySlashCommandHandler(SocketInteraction arg)
+{
+  switch(arg.Name)
+  {
+    case "ping":
+      await arg.RespondAsync("Pong!");
+      break;
+  }
+}
+```
+
+#### Example hug command
+```cs
+_client.InteractionCreated += MyEventHandler;
+_client.Ready += CreateCommands;
+
+...
+
+private async Task CreateCommands()
+{
+  await _client.Rest.CreateGlobalCommand(new Discord.SlashCommandCreationProperties()
+  {
+    Name = "hug",
+    Description = "Hugs a user!",
+    Options = new List<Discord.ApplicationCommandOptionProperties>()
+    {
+      new ApplicationCommandOptionProperties()
+      {
+        Name = "User",
+        Required = true,
+        Description = "The user to hug",
+        Type = Discord.ApplicationCommandOptionType.User,
+      }
+    }
+  });
+}
+
+private async Task MyEventHandler(SocketInteraction arg)
+{
+  switch(arg.Type) // We want to check the type of this interaction
+  {
+    case InteractionType.ApplicationCommand: // If it is a command
+      await MySlashCommandHandler(arg); // Handle the command somewhere
+      break;
+     default: // We dont support it
+      Console.WriteLine("Unsupported interaction type: " + arg.Type);
+      break;
+  }
+}
+
+private async Task MySlashCommandHandler(SocketInteraction arg)
```

Comment:
Remove this parameter and make it a variable in like `SlashModuleBase<BaseSlashCommandContext>` (don't omit the generic because it's a really nice addition to modify the context to fit your needs (as I do in my project))